test/common/shared_functions.go

```diff
@@ -217,3 +220,65 @@ func WriteRHMICRToFile(client dynclient.Client, file string) error {
 		return writeObjToYAMLFile(rhmi, file)
 	}
 }
+
+// Common function to perform CRUDL and verifying their expected permissions
+func verifyCRUDLPermissions(t *testing.T, openshiftClient *resources.OpenshiftClient, expectedPermission ExpectedPermissions) {
+	// Perform LIST Request
+	resp, err := openshiftClient.DoOpenshiftGetRequest(expectedPermission.ListPath)
+
+	if err != nil {
+		t.Errorf("failed to perform LIST request with error : %s", err)
+	}
+
+	if resp.StatusCode != expectedPermission.ExpectedListStatusCode {
+		t.Errorf("unexpected response from LIST request : %v", resp)
+	}
+
+	// Perform CREATE Request
+	bodyBytes, err := json.Marshal(expectedPermission.ObjectToCreate)
+
+	resp, err = openshiftClient.DoOpenshiftPostRequest(expectedPermission.ListPath, bodyBytes)
+
+	if err != nil {
+		t.Errorf("failed to perform CREATE request with error : %s", err)
+	}
+
+	if resp.StatusCode != expectedPermission.ExpectedCreateStatusCode {
+		t.Errorf("unexpected response from CREATE request : %v", resp)
+	}
+
+	// Perform GET Request
+	resp, err = openshiftClient.DoOpenshiftGetRequest(expectedPermission.GetPath)
+
+	if err != nil {
+		t.Errorf("failed to perform GET request with error : %s", err)
+	}
+
+	if resp.StatusCode != expectedPermission.ExpectedReadStatusCode {
+		t.Errorf("unexpected response from GET request : %v", resp)
+	}
+
+	// Perform UPDATE Request
+	bodyBytes, err = ioutil.ReadAll(resp.Body) // Use response from GET
```

Comment:
Should we be closing `resp.Body` at some point?

Reply:
Also same as above, would be nice to handle this error

Reply:
Ah yes, good catch , probably be a good idea to close the `resp.body` and handle the error :+1: